torch/csrc/distributed/c10d/init.cpp

```diff
@@ -417,11 +417,17 @@ They are used in specifying strategies for reduction collectives, e.g.,
 #endif
 
 #ifdef USE_C10D_MPI
-  shared_ptr_class_<::c10d::ProcessGroupMPI>(
-      module, "ProcessGroupMPI", processGroup)
-      .def(py::init([](std::vector<int> ranks) {
+  auto processGroupMPI = shared_ptr_class_<::c10d::ProcessGroupMPI>(
+      module, "ProcessGroupMPI", processGroup);
+
+  // Define static create function instead of a constructor, because
+  // this function may return null. This happens if this process is not
+  // part of a sub group that is to be created.
+  processGroupMPI.def_static(
```

Comment:
The existing interface is not great, with the leakage of global rank and group rank into the process group class here. I would be surprised if anybody uses this interface directly. Strictly speaking, they could be, and this could be a concern, but I think it's safe to change the interface here.